validations/src/tests.rs

```diff
@@ -3361,6 +3445,254 @@ fn block_add_vtt_but_dont_update_merkle_tree() {
     );
 }
 
+#[test]
+fn block_duplicated_commits() {
+    let mut dr_pool = DataRequestPool::default();
+    let vrf = &mut VrfCtx::secp256k1().unwrap();
+
+    let secret_key = SecretKey {
+        bytes: Protected::from(vec![0xcd; 32]),
+    };
+    let current_epoch = 1000;
+    let last_block_hash = LAST_BLOCK_HASH.parse().unwrap();
+    let block_beacon = CheckpointBeacon {
+        checkpoint: current_epoch,
+        hash_prev_block: last_block_hash,
+    };
+    // Add commits
+    let commit_beacon = block_beacon;
+
+    let dro = DataRequestOutput {
+        value: 1000,
+        commit_fee: 50,
+        witnesses: 2,
+        data_request: example_data_request(),
+        ..DataRequestOutput::default()
+    };
+    let dr_body = DRTransactionBody::new(vec![], vec![], dro);
+    let drs = sign_t(&dr_body);
+    let dr_transaction = DRTransaction::new(dr_body, vec![drs]);
+    let dr_hash = dr_transaction.hash();
+    let dr_epoch = 0;
+    dr_pool
+        .process_data_request(
+            &dr_transaction,
+            dr_epoch,
+            EpochConstants::default(),
+            &Hash::default(),
+        )
+        .unwrap();
+
+    // Insert valid proof
+    let mut cb = CommitTransactionBody::default();
+    cb.dr_pointer = dr_hash;
+    cb.proof =
+        DataRequestEligibilityClaim::create(vrf, &secret_key, commit_beacon, dr_hash).unwrap();
+    // Sign commitment
+    let cs = sign_t(&cb);
+    let c_tx = CommitTransaction::new(cb, vec![cs]);
+
+    let mut block_hash = Hash::default();
+    let x = test_block_with_drpool(
+        |b| {
+            // We include a duplicate tally
+            b.txns.commit_txns.push(c_tx.clone());
+            b.txns.commit_txns.push(c_tx.clone());
```

Comment:
Aside from pushing the same commit twice, could you also test two different commits from the same pkh?

    let mut cb2 = cb.clone();
    cb2.commitment = Hash::SHA256([1; 32]);
    let cs = sign_t(&cb2);
    let c2_tx = CommitTransaction::new(cb2, vec![cs2]);